libraries/chain/asset_evaluator.cpp

```diff
@@ -755,7 +754,19 @@ operation_result asset_settle_evaluator::do_apply(const asset_settle_evaluator::
             obj.settlement_fund -= settled_amount.amount;
          });
 
-         d.adjust_balance( op.account, settled_amount );
+         // The account who settles pays market fees to the issuer of the collateral asset after HF core-1780
+         //
+         // TODO Check whether the HF check can be removed after the HF.
+         //      Note: even if logically it can be removed, perhaps the removal will lead to a small
+         //            performance loss. Needs testing.
+         if( d.head_block_time() >= HARDFORK_CORE_1780_TIME )
+         {
+            auto issuer_fees = d.pay_market_fees( fee_paying_account, settled_amount.asset_id(d), settled_amount );
```

Comment:
@abitmore  This invocation of `pay_market_fees` from the evaluator is difficult to resolve and merge with another PR from simple maker-taker fees from BSIP81 because the market fee will be dependent on whether the order is considered to be a maker or taker.

I am attempting to find a reasonable manner to merge these two PRs.  Because this line of logic applies during a global settlement where the settlement pool is waiting, is it reasonable to consider this settlement order as a taker?

Reply:
Yes, the settlement order is always taker.

Reply:
Actually, the settlement order can be maker sometimes if we implemented bsip73.

Reply:
A side note, since there is no virtual `fill_order_operation` generated for instant settlement, neither the price nor the volume would be in the HTLCV charts.

Reply:
If I merge this before the BSIP73 PR, can I ignore the problem? :)

Reply:
Sure. I believe 73 will be worked on only after this.